src/Parsers.jl

```diff
@@ -170,6 +170,62 @@ Options(;
 const OPTIONS = Options(nothing, UInt8(' '), UInt8('\t'), UInt8('"'), UInt8('"'), UInt8('"'), nothing, UInt8('.'), nothing, nothing, nothing, false, false, nothing, false, false, false, nothing)
 const XOPTIONS = Options(missing, UInt8(' '), UInt8('\t'), UInt8('"'), UInt8('"'), UInt8('"'), UInt8(','), UInt8('.'), nothing, nothing, nothing, false, false, nothing, true, false, false, nothing)
 
+abstract type AbstractTypeParseConf{T} end
+abstract type AbstractDateTimeParseConf{T} <: AbstractTypeParseConf{T} end
+abstract type AbstractFloatParseConf{T}    <: AbstractTypeParseConf{T} end
+abstract type AbstractIntParseConf{T}      <: AbstractTypeParseConf{T} end
+abstract type AbstractStringParseConf{T}   <: AbstractTypeParseConf{T} end
```

Comment:
Any reason not to have an AbstractBoolParseConf?

Reply:
I guess we don't really expect custom user-defined alternative `Bool` types to exist, so probably fine w/o.

Reply:
What are these `AbstractXParserConf`s giving us? It looks like only `AbstractStringParseConf` is used (unless i missed some), so i'm wondering if we need these other ones at all

Or perhaps to ask the question another way: if someone were to define there own bool, e.g. `MyTrue <: MyBool <: Integer`, what different would there be in them defining a `MyBoolParseConf` that is `<: AbstractIntParseConf` versus `<: AbstractBoolParseConf` versus `<: AbstractTypeParseConf`?

Reply:
> I guess we don't really expect custom user-defined alternative Bool types to exist, so probably fine w/o.

Yeah, there is no `AbstractBool`, so I didn't bother with `AbstractBoolParseConf`. Not sure if we need it, but maybe it would be more surprising to users to exclude it?

> What are these AbstractXParserConfs giving us?

A good question. The answer is not much. The idea was that if someone wants to parse their own Integer (like `SafeIntegers.jl`) they could define their own `Conf` type as a subtype of `AbstractIntParseConf` then they don't have to define their own parsing method. For `FixedPointDecimals.jl` one would definitely need to provide a custom parsing method, since there is nothing compatible in `Parsers.jl`, so in this case, there is no benefit subtype anything more specific than `AbstractTypeParseConf`.